container/create_image_config.py

```diff
@@ -121,10 +125,30 @@ def Stamp(inp):
     elif '{' in value:
       labels[label] = Stamp(value)
 
+  creation_time = None
+  if args.creation_time:
+    try:
+      # If creation_time is parsable as an int, assume unix epoch timestamp.
+      unix_timestamp = int(args.creation_time)
+
+      # Construct a RFC 3339 date/time from the Unix epoch.
+      creation_time = (
+          datetime.datetime.utcfromtimestamp(
+              unix_timestamp
+          ).strftime("%Y-%m-%dT%H:%M:%SZ")
+      )
+    except ValueError:
+      # Otherwise, assume RFC 3339 date/time format.
+      creation_time = args.creation_time
+
   output = v2_2_metadata.Override(data, v2_2_metadata.Overrides(
-      author='Bazel', created_by='bazel build ...',
-      layers=layers, entrypoint=list(map(Stamp, fix_dashdash(args.entrypoint))),
-      cmd=list(map(Stamp, fix_dashdash(args.command))), user=Stamp(args.user),
+      author='Bazel',
+      created_by='bazel build ...',
+      layers=layers,
+      entrypoint=list(map(Stamp, fix_dashdash(args.entrypoint))),
+      cmd=list(map(Stamp, fix_dashdash(args.command))),
+      creation_time=creation_time,
```

Comment:
This doesn't fix the issue unless users can use: `{SOURCE_DATE_EPOCH}` (or w/e the magic stamp variable @ixdy wants is) for this value.

Reply:
They can, kind of, though `$(SOURCE_DATE_EPOCH)`  is seconds since the unix epoch and BUILD_TIMESTAMP is in milliseconds.

Reply:
right, as I understand it, `BUILD_TIMESTAMP` is normally set to the current time, but if `SOURCE_DATE_EPOCH` is set, Bazel sets `BUILD_TIMESTAMP` to that value instead.

tl;dr I think this is doing what we want

Reply:
(it's confusing because they're both yelling case, but `SOURCE_DATE_EPOCH` is an environment variable Bazel recognizes, but it isn't set in the workspace status; only `BUILD_TIMESTAMP` is (unless you set it yourself))